simln-lib/src/eclair.rs

```diff
@@ -0,0 +1,365 @@
+use crate::{
+    serializers, LightningError, LightningNode, NodeId, NodeInfo, PaymentOutcome, PaymentResult,
+};
+use async_trait::async_trait;
+use bitcoin::secp256k1::PublicKey;
+use bitcoin::Network;
+use lightning::ln::features::NodeFeatures;
+use lightning::ln::{PaymentHash, PaymentPreimage};
+use reqwest::multipart::Form;
+use reqwest::{Client, Method, Url};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::collections::HashMap;
+use std::str::FromStr;
+use std::time::Duration;
+use tokio::time;
+use triggered::Listener;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct EclairConnection {
+    #[serde(with = "serializers::serde_node_id")]
+    pub id: NodeId,
+    #[serde(with = "serializers::serde_address")]
+    pub base_url: String,
+    pub api_username: String,
+    pub api_password: String,
+}
+
+impl EclairConnection {
+    fn construct_url(&self, endpoint: &str) -> anyhow::Result<Url> {
+        Ok(Url::parse(&self.base_url)?.join(endpoint)?)
+    }
+}
+
+pub struct EclairNode {
+    connection: EclairConnection,
+    http_client: Client,
+    info: NodeInfo,
+    network: Network,
+}
+
+impl EclairNode {
+    async fn request_static<T: for<'de> Deserialize<'de>>(
+        connection: &EclairConnection,
+        client: &Client,
+        endpoint: &str,
+        params: Option<HashMap<String, String>>,
+    ) -> anyhow::Result<T> {
+        let url = connection.construct_url(endpoint)?;
+        let mut request = client
+            .request(Method::POST, url)
+            .basic_auth(&connection.api_username, Some(&connection.api_password));
+
+        if let Some(params) = params {
+            let mut form = Form::new();
+            for (key, value) in params {
+                form = form.text(key, value);
+            }
+            request = request.multipart(form);
+        }
+
+        let response = request
+            .send()
+            .await?
+            .error_for_status()?
+            .json::<T>()
+            .await?;
+
+        Ok(response)
+    }
+
+    async fn request<T: for<'de> Deserialize<'de>>(
```

Comment:
The division in duties of this wrapping function and `request_static` weren't immediately obvious to be, could we either add a comment to explain the need for this or move the connection details into its own `client` struct and implement `request` on it? 

Could work well with the request above, `EclairNode` then just holds a `client` / `info` / `network` - this is how we have it for `LndNode`, for example.

Reply:
`request_static` is used in the constructor, this is the only reason why it exists. Otherwise, it will be only `request`. After all the refactoring it looks a bit cleaner so I won't go deeper for now and leave it as it is. If after second review you still find this can be improved, please let me know and I'll handle it.

Reply:
I was thinking something like [this patch](https://github.com/user-attachments/files/19129979/eclair_client.patch) to pull the request details out into their own struct, so that the `EclairNode`'s single responsibility is implementing `LightningNode` trait.

Reply:
makes sense! Done https://github.com/bitcoin-dev-project/sim-ln/pull/225/commits/0157beb38b659f5afedc0e22ac79a39d87f36a57